src/pickBy.ts

```diff
@@ -0,0 +1,47 @@
+import { purry } from './purry';
+
+/**
+ * Returns a partial copy of an object containing only the keys that satisfy the supplied predicate.
+ * @param object the target object
+ * @param pred predicate
+ * @signature R.pickBy(object, (value, key) => true )
+ * @example
+ *    R.pickBy({ a: 1, b: 2, c: 3, d: 4 }, (value) => value === 1 || value === 4) // => { a: 1, d: 4 }
+ * @data_first
+ * @category Object
+ */
+export function pickBy<T extends object>(
+  object: T,
+  pred: (value: any, key: string) => boolean
+): Partial<T>;
```

Comment:
Partial is definitely a very wide type. It would be good if the return type could be strictly inferred from what the predicate function returns. So if it returns `true` for a specific key, it will be included in the result.